### PR TITLE
Rename controller to strategy

### PR DESCRIPTION
**Pull request by jpkrohling**
Signed-off-by: Juraci Paixão Kröhling <juraci@kroehling.de>

### CODE REVIEW

**Comment by jpkrohling** [comment]
This change is [Reviewable](https://reviewable.io/reviews/jaegertracing/jaeger-operator/125)

**Comment by jpkrohling** [comment]
Summary of changes:

1. Moved the old controller code to `strategy`
1. Renamed controllers to `strategy` (`allInOneController` -> `allInOneStrategy`)
1. Renamed the interface `Controller` to `S`, referenced by code in other packages as `strategy.S`
1. Renamed `NewController` to `For`, referenced by code in other packages as `strategy.For(ctx, jaeger)`

**Comment by objectiser** [comment]
@jpkrohling Before doing the review - don't like `S` as an interface name - could it be changed to `Strategy`, or if don't want `strategy.Strategy`, possibly `Plan` (i.e. the outcome of a strategy)?

**Comment by jpkrohling** [comment]
> don't like S as an interface name

I'm not a big fan either and I did name it `strategy.Strategy` first, but I changed it to follow Go conventions in naming the type with a single letter when the type has the same name as the package. `testing.T` is the most common reference of this rule.

File: pkg/strategy/controller_test.go [hunk 1]
```diff
@@ -179,7 +179,7 @@ func getDeployments(objs []runtime.Object) []*appsv1.Deployment {
 	return deps
 }
 
-func assertHasAllObjects(t *testing.T, name string, objs []runtime.Object, deployments map[string]bool, daemonsets map[string]bool, services map[string]bool, ingresses map[string]bool, routes map[string]bool, serviceAccounts map[string]bool) {
+func assertHasAllObjects(t *testing.T, name string, objs []runtime.Object, deployments map[string]bool, daemonsets map[string]bool, services map[string]bool, ingresses map[string]bool, routes map[string]bool, serviceAccounts map[string]bool, configMaps map[string]bool) {
```
Comment (objectiser) [inline pkg/strategy/controller_test.go]: Not necessarily for now, but could probably do with a struct to contain all of these parameters.

File: pkg/strategy/production_test.go [hunk 2]
```diff
@@ -45,14 +45,29 @@ func TestCreateProductionDeploymentWithDaemonSetAgent(t *testing.T) {
...
 	j.Spec.Agent.Strategy = "DaemonSet"
 
-	c := newProductionController(context.TODO(), j)
+	c := newProductionStrategy(context.TODO(), j)
 	objs := c.Create()
-	assertDeploymentsAndServicesForProduction(t, name, objs, true, false)
+	assertDeploymentsAndServicesForProduction(t, name, objs, true, false, false)
+}
+
+func TestCreateProductionDeploymentWithUIConfigMap(t *testing.T) {
```
Comment (objectiser) [inline pkg/strategy/production_test.go]: Sorry I missed these :)

**Pull request merged**